pkg/controlplane/reconcilers/lease.go

```diff
@@ -120,7 +120,8 @@ func (s *storageLeases) UpdateLease(ip string) error {
 
 // RemoveLease removes the lease on a master IP in storage
 func (s *storageLeases) RemoveLease(ip string) error {
-	return s.storage.Delete(apirequest.NewDefaultContext(), s.baseKey+"/"+ip, &corev1.Endpoints{}, nil, rest.ValidateAllObjectFunc, nil)
+	key := path.Join(s.baseKey, ip)
```

Comment:
Can you avoid using path.Join for this? I know it's a pain but hiding the call to path.Clean is a problem. 

cc @tallclair

Reply:
at head, in a followup, sure. For backports, I don't think it's worth it. baseKey is hardcoded and IP is a stringified net.IP so it can't contain weird slash/space/backstep chars

Reply:
Sounds good.